py-rattler/rattler/package/package_name.py

```diff
@@ -71,9 +71,67 @@ def normalized(self) -> str:
         """
         return self._name.normalized
 
+    def __hash__(self) -> int:
+        """
+        Computes the hash of this instance.
+
+        Examples
+        --------
+        ```python
+        >>> hash(PackageName("test-abc")) == hash(PackageName("test-abc"))
+        True
+        >>> hash(PackageName("test-abc")) == hash(PackageName("abc-test"))
```

Comment:
If this wraps the rattler Package name, I think one of the key features is that it's case-insensitive.

So maybe checking that `hash(PackageName("test-abc")) == hash(PackageName("test-ABC"))`   would be a better check (same for equality).